### PR TITLE
Add test cases for numbers passed as categorical attributes

### PR DESCRIPTION
**Pull request by aarsilv**
Adds test cases that cover:
* Number passed as a subject categorical attribute
* Non-number passed as a subject numeric attribute
* Number passed as an action categorical attribute
* Non-number passed as an action numeric attribute

The tests are designed with assignments close to the boundary of the action weights so that if the numbers are not treated as categorical attributes and weighted appropriately, the tests will fail.

### CODE REVIEW

File: ufc/bandit-flags-v1.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
                         "variationKey": "control",
                         "shards": []
                     }
-                ]
+                ],
+                "doLog": true
```
Comment (aarsilv) [inline ufc/bandit-flags-v1.json:21]: Adding in the missing `doLog`s

File: ufc/bandit-tests/test-case-bandit-does-not-exist.json [hunk 2]
```diff
@@ -4,9 +4,9 @@
     "subjects": [
       {
         "subjectKey": "alice",
-        "subjectAttributes": {"numeric_attributes": {"age": 25}, "categorical_attributes": {"country": "USA"}},
+        "subjectAttributes": {"numericAttributes": {"age": 25}, "categoricalAttributes": {"country": "USA"}},
```
Comment (aarsilv) [inline ufc/bandit-tests/test-case-bandit-does-not-exist.json:7]: renamed from 🐍  to 🐫 

Note: this _will_ break all existing SDK tests 😞

File: ufc/bandit-tests/test-case-banner-bandit.json [hunk 3]
```diff
@@ -211,6 +211,56 @@
             }
         ],
         "assignment": {"variation": "banner_bandit", "action": "nike"}
+      },
+      {
+        "subjectKey": "henry",
+        "subjectAttributes": {
+          "numericAttributes": { "age": 25, "mistake": "oops" },
+          "categoricalAttributes": {"country": "USA", "gender_identity": "female", "area_code": 303}
```
Comment (aarsilv) [inline ufc/bandit-tests/test-case-banner-bandit.json:219]: `area_code` pushes Henry into the adidas bucket

File: ufc/bandit-tests/test-case-banner-bandit.json [hunk 4]
```diff
@@ -211,6 +211,56 @@
             }
         ],
         "assignment": {"variation": "banner_bandit", "action": "nike"}
+      },
+      {
+        "subjectKey": "henry",
+        "subjectAttributes": {
+          "numericAttributes": { "age": 25, "mistake": "oops" },
```
Comment (aarsilv) [inline ufc/bandit-tests/test-case-banner-bandit.json:218]: `mistake` should be ignored (vs. throwing an error)

File: ufc/bandit-tests/test-case-banner-bandit.json [hunk 5]
```diff
@@ -211,6 +211,56 @@
...
+      {
+        "subjectKey": "imogene",
+        "subjectAttributes": {
+          "numericAttributes": { "age": 25},
+          "categoricalAttributes": {"country": "USA", "gender_identity": "female"}
+        },
+        "actions": [
+          {
+            "actionKey": "nike",
+            "numericAttributes": {"brand_affinity": -1.7, "mistake": true},
```
Comment (aarsilv) [inline ufc/bandit-tests/test-case-banner-bandit.json:249]: Just like the subject attribute, `mistake` should be ignored (vs. throwing an error)

File: ufc/bandit-tests/test-case-banner-bandit.json [hunk 6]
```diff
@@ -211,6 +211,56 @@
...
+            "categoricalAttributes": {"loyalty_tier": "silver", "zip":  81427}
```
Comment (aarsilv) [inline ufc/bandit-tests/test-case-banner-bandit.json:250]: `zip` should push Imogene into the Nike bucket

**Review by typotter — APPROVED** [review]
Thanks for the 🐍 fix and the updated test cases. Do you think it's worth having a test case file specifically for some of these corner case inputs? I'm thinking that reconstructing the codepaths these test cases aim to follow is becoming somewhat tribal knowledge. Maybe we could add a note/comment field to the `TestCaseSubject` object. Wdyt?

File: ufc/bandit-tests/test-case-banner-bandit.json (hunk 4 above)
Comment (aarsilv) [reply ufc/bandit-tests/test-case-banner-bandit.json:218]: Great call out yes I think it's warning worthy

Comment (giorgiomartini0) [reply ufc/bandit-tests/test-case-banner-bandit.json:218]: It's not obvious to me that this is warning-worthy. It would trigger warnings every time a new attribute is used, in particular for _all_ attributes during the cold start phase (when the bandit model is empty or missing). Info-level logging, sure: "hey we found a new attribute". These go away automatically after a new bandit training run, when the attribute will now be part of the model. New attributes are not necessarily indicative of a problem.

**Review by giorgiomartini0 — APPROVED** [review]

File: ufc/bandit-tests/test-case-banner-bandit.json (hunk 4 above)
Comment (typotter) [reply ufc/bandit-tests/test-case-banner-bandit.json:218]: It's not that they're ignored by the model - it's that they're ignored as input at the SDK layer because they've been passed as the wrong attribute type (a string value listed with the numeric attributes). Strings passed specifically as "numeric" attributes are dropped when model evaluation happens. It means that the developer has made a mistake in classifying subject/action attributes in their own code/data and should fix it.

Comment (giorgiomartini0) [reply ufc/bandit-tests/test-case-banner-bandit.json:218]: You are very right! I missed the (now obvious) type mismatch. Agree that it should be a warning.
